Running MergeAppBundles task on arm64 dotnet fails

```
/usr/local/share/dotnet/packs/Microsoft.macOS.Sdk/12.0.100-preview.7.230/targets/Xamarin.Shared.Sdk.targets(308,3): error MSB4018: The "MergeAppBundles" task failed unexpectedly. [/Users/filipnavara/Projects/emclient/MailClient/MailClient.csproj]
/usr/local/share/dotnet/packs/Microsoft.macOS.Sdk/12.0.100-preview.7.230/targets/Xamarin.Shared.Sdk.targets(308,3): error MSB4018: System.EntryPointNotFoundException: Unable to find an entry point named 'lstat$INODE64' in shared library '/usr/lib/libSystem.dylib'. [/Users/filipnavara/Projects/emclient/MailClient/MailClient.csproj]
/usr/local/share/dotnet/packs/Microsoft.macOS.Sdk/12.0.100-preview.7.230/targets/Xamarin.Shared.Sdk.targets(308,3): error MSB4018:    at Xamarin.Utils.PathUtils.lstat(String path, stat& buf) [/Users/filipnavara/Projects/emclient/MailClient/MailClient.csproj]
/usr/local/share/dotnet/packs/Microsoft.macOS.Sdk/12.0.100-preview.7.230/targets/Xamarin.Shared.Sdk.targets(308,3): error MSB4018:    at Xamarin.Utils.PathUtils.IsSymlink(String file) in /Users/builder/azdo/_work/1/s/xamarin-macios/tools/common/PathUtils.cs:line 196 [/Users/filipnavara/Projects/emclient/MailClient/MailClient.csproj]
/usr/local/share/dotnet/packs/Microsoft.macOS.Sdk/12.0.100-preview.7.230/targets/Xamarin.Shared.Sdk.targets(308,3): error MSB4018:    at Xamarin.MacDev.Tasks.MergeAppBundlesTaskBase.GetFileType(String path) in /Users/builder/azdo/_work/1/s/xamarin-macios/msbuild/Xamarin.MacDev.Tasks.Core/Tasks/MergeAppBundlesTaskBase.cs:line 448 [/Users/filipnavara/Projects/emclient/MailClient/MailClient.csproj]
/usr/local/share/dotnet/packs/Microsoft.macOS.Sdk/12.0.100-preview.7.230/targets/Xamarin.Shared.Sdk.targets(308,3): error MSB4018:    at Xamarin.MacDev.Tasks.MergeAppBundlesTaskBase.MergeAppBundles() in /Users/builder/azdo/_work/1/s/xamarin-macios/msbuild/Xamarin.MacDev.Tasks.Core/Tasks/MergeAppBundlesTaskBase.cs:line 287 [/Users/filipnavara/Projects/emclient/MailClient/MailClient.csproj]
/usr/local/share/dotnet/packs/Microsoft.macOS.Sdk/12.0.100-preview.7.230/targets/Xamarin.Shared.Sdk.targets(308,3): error MSB4018:    at Xamarin.MacDev.Tasks.MergeAppBundlesTaskBase.Execute() in /Users/builder/azdo/_work/1/s/xamarin-macios/msbuild/Xamarin.MacDev.Tasks.Core/Tasks/MergeAppBundlesTaskBase.cs:line 249 [/Users/filipnavara/Projects/emclient/MailClient/MailClient.csproj]
/usr/local/share/dotnet/packs/Microsoft.macOS.Sdk/12.0.100-preview.7.230/targets/Xamarin.Shared.Sdk.targets(308,3): error MSB4018:    at Xamarin.MacDev.Tasks.MergeAppBundles.Execute() in /Users/builder/azdo/_work/1/s/xamarin-macios/msbuild/Xamarin.MacDev.Tasks/Tasks/MergeAppBundles.cs:line 10 [/Users/filipnavara/Projects/emclient/MailClient/MailClient.csproj]
/usr/local/share/dotnet/packs/Microsoft.macOS.Sdk/12.0.100-preview.7.230/targets/Xamarin.Shared.Sdk.targets(308,3): error MSB4018:    at Microsoft.Build.BackEnd.TaskExecutionHost.Microsoft.Build.BackEnd.ITaskExecutionHost.Execute() [/Users/filipnavara/Projects/emclient/MailClient/MailClient.csproj]
/usr/local/share/dotnet/packs/Microsoft.macOS.Sdk/12.0.100-preview.7.230/targets/Xamarin.Shared.Sdk.targets(308,3): error MSB4018:    at Microsoft.Build.BackEnd.TaskBuilder.ExecuteInstantiatedTask(ITaskExecutionHost taskExecutionHost, TaskLoggingContext taskLoggingContext, TaskHost taskHost, ItemBucket bucket, TaskExecutionMode howToExecuteTask) [/Users/filipnavara/Projects/emclient/MailClient/MailClient.csproj]
```

--- a/tools/common/PathUtils.cs
+++ b/tools/common/PathUtils.cs
@@ -187,8 +187,20 @@ namespace Xamarin.Utils
 			public ulong st_qspare_2;
 		}
 
-		[DllImport ("/usr/lib/libSystem.dylib", EntryPoint = "lstat$INODE64", SetLastError = true)]
-		static extern int lstat (string path, out stat buf);
+		[DllImport ("/usr/lib/libc.dylib", EntryPoint = "lstat$INODE64", SetLastError = true)]
+		static extern int lstat_x64 (string file_name, out stat buf);
+
+		[DllImport ("/usr/lib/libc.dylib", EntryPoint = "lstat", SetLastError = true)]
+		static extern int lstat_arm64 (string file_name, out stat buf);
+
+		static int lstat (string path, out stat buf)
+		{
+			if (RuntimeInformation.ProcessArchitecture == Architecture.Arm64) {
+				return lstat_arm64 (path, out buf);
+			} else {
+				return lstat_x64 (path, out buf);
+			}
+		}
 
 		public static bool IsSymlink (string file)
 		{

--- a/tools/mmp/mmp.csproj
+++ b/tools/mmp/mmp.csproj
@@ -7,7 +7,7 @@
     <OutputType>Exe</OutputType>
     <AssemblyName>mmp</AssemblyName>
     <RootNamespace>mmp</RootNamespace>
-    <TargetFrameworkVersion>v4.6</TargetFrameworkVersion>
+    <TargetFrameworkVersion>v4.8</TargetFrameworkVersion>
     <LangVersion>latest</LangVersion>
   </PropertyGroup>
   <PropertyGroup Condition=" '$(Configuration)|$(Platform)' == 'Debug|AnyCPU' ">

